airflow-core/src/airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -617,14 +617,17 @@ def _evaluate_direct_relatives() -> Iterator[TIDepStatus]:
                 elif success == 0:
                     yield self._failing_status(
                         reason=(
-                            f"Task's trigger rule '{trigger_rule}' requires all non-skipped upstream tasks to have "
+                            f"Task's trigger rule '{trigger_rule.value}' requires all non-skipped upstream tasks to have "
                             f"completed and at least one upstream task has succeeded, but found "
                             f"{success} successful task(s). upstream_states={upstream_states}, "
                             f"upstream_task_ids={task.upstream_task_ids}"
                         )
                     )
             else:
-                yield self._failing_status(reason=f"No strategy to evaluate trigger rule '{trigger_rule}'.")
+                print("true rule not implemented:", trigger_rule)
+                yield self._failing_status(
+                    reason=f"No strategy to evaluate trigger rule '{getattr(trigger_rule, 'value', trigger_rule)}'."
```

Comment:
Why getattr here?

Reply:
Just replaced all the occurecnes with `getattr`, reason being:

* task-sdk TriggerRule has no `__str__`, therefore, f"{trigger_rule}" → "TriggerRule.ALL_SUCCESS"
* airflow-core TriggerRule has `__str__` and it behaves like f"{trigger_rule}" → "all_success"

That's why getattr(trigger_rule, 'value', trigger_rule) works for all cases:
Enum objects: Uses .value => "all_success"
String objects: Falls back to string => "unknown_status"

Reply:
> Also how does this code get run with the sdk version of the enum in the first place?

All the tasks are written using the task flow api and those use TriggerRule enum of the sdk within sdk:
```
self.trigger_rule: TriggerRule = TriggerRule(trigger_rule) 
```

^ in BaseOperator.__init__

Reply:
> What if you change the generation to have the use-subclass-enum option?
I think that will make it subclass string and not need this?

Let me try

Reply:
Resolving to unblock CI runs, can later handle if anything